### DISCUSSION TITLE
section 3 : Latex

### DISCUSSION THREAD

**Discussion by arifulone** [Cyfrin/security-and-auditing-full-course-s23#45] [Q&A] [answered]
i've tried to install Latex on my linux machine using
```bash
 sudo apt-get install texlive-full
```
this package is more than 7 GB in total and  now my system is run out of storage. i have only 120 GB of ssd system. 
so, i remove it . 

can u guys give a link where i can find small LaTex package which is enough to job done ? (less than 1 GB).

**Comment by PatrickAlphaC** [comment]
Hmm.... Could you try a cloud option perhaps? Like setup a VM in the cloud and run it there? 

otherwise... you could also skip creating the PDF

**Comment by johnsonstephan** [comment]
Hi @arifulone, I totally understand your situation! I had a similar issue while running my Linux VM. The `texlive-full` package can indeed be quite large and a great alternative is TinyTeX. 

**TL;DR: Here is a link to [TinyTeX](https://github.com/rstudio/tinytex), a small LaTex package which is 66MB.**

---

TinyTeX is a very light solution, which will save you from running out of storage. It is also user-friendly, especially if you're just getting started with LaTeX. 

<details><summary>Installation Instructions </summary>
<p>

1. Open your Linux terminal.
2. Run the following command:

   ```bash
   wget -qO- "https://yihui.org/tinytex/install-bin-unix.sh" | sh
   ```

This command will download and install TinyTeX. You may find more details on the package in its [documentation](https://yihui.org/tinytex/).
</p>
</details> 

<details><summary>Bonus Information</summary>
<p>

1. I noticed @PatrickAlphaC suggested using a cloud VM which is a great route to go (for example, [EC2 from AWS](https://aws.amazon.com/getting-started/launch-a-virtual-machine-B-0/)) if you want to use the `texlive-full` package with your setup. However, if you're set on having LaTeX locally, TinyTeX will solve your issue.
2. If you're looking for a fun OS, I love [ZIION](https://www.ziion.org) because it's specifically tailored for smart contract security and is pre-loaded with a plethora of tools, from decompilers to vulnerability scanners. 

</p>
</details> 



Does this help to solve your question?

**Comment by sakarkc2122** [comment]
You can check this [TeX Live - Quick install for Unix](https://www.tug.org/texlive/quickinstall.html).
There you can find a `Changing defaults` section where you can download the relatively smaller version of tex-live using the command-line options mentioned there.

**Comment by johnsonstephan** [comment]
@arifulone Were you finally able to find a solution?

**Comment by sakarkc2122** [comment, marked as answer]
## Latex working for Windows 11 WSL Ubuntu:
#### To uninstall the TeX Live package. The * character is a wildcard that matches all packages that start with "texlive".
1. `sudo apt-get purge texlive-*`
#### Remove unused packages and clean the package cache
2. `sudo apt-get autoremove`
3. `sudo apt-get autoclean`

#### Remove texlive folder
1. `cd ~/../../usr/local/`
2. `sudo rm -rf texlive/` # this removes texlive folder. Remove is irreversible. So, be careful.

#### If you already have the `texmf` folder in the home directory, remove that too. You should be able to do that. Similar to the last command.

### Installation
#### visit [tug-Quick Installation for Unix](https://tug.org/texlive/quickinstall.html). Or Follow these:
1. cd /tmp # working directory of your choice
2. `wget https://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz` # or curl instead of wget
3. `zcat < install-tl-unx.tar.gz | tar xf -`
4. cd install-tl-*
5. `perl ./install-tl --no-interaction` # as root or with writable destination  # This line download +7GB. So please go ahead and skip to next command if storage is the issue.
5. `sudo perl ./install-tl --no-interaction --scheme=small --no-doc-install --no-src-install` # if guide is needed, check quick install above. This took about 11 minutes for me.

**Note:** Finally, prepend /usr/local/texlive/YYYY/bin/PLATFORM to your PATH, e.g., /usr/local/texlive/2023/bin/x86_64-linux
like this: 
1. `nano ~/.bashrc` => this opens a nano text editor. learn to use this editor
2. Go all the way down using the down arrow. The add:
```javascript
export PATH="/usr/local/texlive/2023/bin/x86_64-linux:$PATH"
export MANPATH="$MANPATH:/usr/local/texlive/2023/texmf-dist/doc/man"
export INFOPATH="$INFOPATH:/usr/local/texlive/2023/texmf-dist/doc/info"
```

Now you can use tlmgr

Assuming you got the report.md written down.
Next, save that eisvogel.latex file anywhere you wish. But your command would be:
`pandoc report.md -o report.pdf --from markdown --template=/*location-to-eisvogel*/eisvogel --listings`

If this shows something like: "File 'sourcesanspro.sty' not found. Then:
1. `tlmgr init-usertree` # Sets up a texmf tree
2. I have to use --usermode for every install like:
    `tlmgr --usermode install sourcesanspro`

Note: I needed these installations: footnotebackref titling csquotes mdframed zref needspace sourcesanspro ly1 sourcecodepro. Just run that pandoc command, check if any files are missing, and install them using `tlmgr --usermode install $package_name`.
Thats it.

**Reply by cromewar** [reply]
Amazing answer @sakarkc2122

**Reply by tobezhanabi** [reply]
let me give this a try

**Reply by sharonphiliplima** [reply]
I had to install `mdframed` extra

**Reply by EngrPips** [reply]
Glad you figured that out.
